src/Controls/tests/Xaml.UnitTests/OnPlatformOptimization.xaml.cs

```diff
@@ -31,6 +30,18 @@ public void OnPlatformExtensionsAreSimplified([Values("net6.0-ios", "net6.0-andr
 			{
 				MockCompiler.Compile(typeof(OnPlatformOptimization), out var methodDef, targetFramework);
 				Assert.That(!methodDef.Body.Instructions.Any(instr => InstructionIsOnPlatformExtensionCtor(methodDef, instr)), "This Xaml still generates a new OnPlatformExtension()");
+
+				var expected = targetFramework == "net6.0-ios" ? "bar" : "foo";
+				Assert.That(methodDef.Body.Instructions.Any(instr => instr.Operand as string == expected), $"Did not find instruction containing '{expected}'");
```

Comment:
Maybe this should be a new/separate test so we can check what is failing?

Reply:
Is the assertion message not enough? `MockCompiler.Compile()` actually runs XamlC and generates a `.dll`. I'm not sure how slow that part is, we'd run it twice?